### DISCUSSION TITLE
Fixing automatic tpm2 unlock

### DISCUSSION THREAD

**Discussion by J-Kappes** [szaimen/Nextcloud-NAS-Guide#132] [Q&A] [answered]
I don't know why, but the automatic tpm2 unlock has stopped working. Right before the crypt key prompt, I get the message that the "ubuntu-vg" volume isn't found.

After manually entering my crypt key, everything seems to run just fine.

I tried running tpm2-unlock.sh again, but it says it shouldn't be run twice.

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/vm/pull/2692

**Reply by J-Kappes** [reply]
The script now exits with "No TPM 2.0 device found." and the server still doesn't automatically unlock on reboot.

**Reply by szaimen** [reply]
So something is wrong on your server as there is apparently no tpm2 device anymore?

**Reply by szaimen** [reply]
See https://github.com/nextcloud/vm/blob/472e5f2ab216ef2b1de4ab25657a00cc85b4c5d5/not-supported/tpm2-unlock.sh#L47-L49

**Reply by szaimen** [reply]
You can check yourself with `sudo dmesg | grep -i "tpm" `

**Reply by J-Kappes** [reply]
Or the outputs got changed to no longer include "2.0":
```
sudo dmesg | grep -i "tpm"
[    0.000000] efi: TPMFinalLog=0x78d84000 ACPI=0x78e2b000 ACPI 2.0=0x78e2b014 SMBIOS=0x75e53000 SMBIOS 3.0=0x75e51000 ESRT=0x75e3e398 MEMATTR=0x72ef7018 MOKvar=0x75e36000 RNG=0x78df5018 TPMEventLog=0x66825018 
[    0.016884] ACPI: TPM2 0x0000000078E26000 000034 (v04 HPQOEM 84D4     00000002 HP   00040000)
[    0.017236] ACPI: Reserving TPM2 table memory at [mem 0x78e26000-0x78e26033]
```

**Reply by szaimen** [reply]
I see. I guess the script needs to be adjusted then. Can you do so?

**Reply by J-Kappes** [reply]
Yes, I've already done so locally. Do you mean here on github?

**Reply by J-Kappes** [reply]
The automatic unlock works now. Thanks :)

**Reply by szaimen** [reply]
> Yes, I've already done so locally. Do you mean here on github?

Yes :)

**Comment by boraozgen** [comment]
My auto unlock broke today after years of use. Re-running the script fixed it, but I don't understand how it could break. What are possible causes? There was a power outage today, can it be related? Can a software update break this? Is there any way to debug this?
